juju/juju
# Action cancel/abort handling on uniter

### Checklist


----

## Action cancel/abort handling on uniter

- Extend k8s exec framework to support signals and cancelation (with
kill).
- Uniter reacts to action change to check for aborting status

## QA steps

Create a charm with an action that sleeps for a few minutes.
Deploy charm.
`juju cancel-action <action-id>`
Action should go to aborting status.
Then should go to aborted status.

## Documentation changes

N/A

## Bug reference

N/A
